api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,432 @@
+     */
+    AnnotatedTypeConfigurator<T> addToMethodParameter(Method method, int position, Annotation annotation);
+
+    /**
+     * Remove an annotation from the specified method parameter.
+     *
+     * @param method         the method to remove the annotation from
+     * @param position       the position of the parameter to remove
+     * @param annotationType the annotation type to remove
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  method is not currently declared on the type or if the
+     *                                  parameter is not declared on the method
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromMethodParameter(Method method,
+                                                           int position, Class<? extends Annotation> annotationType);
+
+    /**
+     * Add an annotation to the specified constructor. If the constructor is not
+     * already present, it will be added.
+     *
+     * @param constructor the constructor to add the annotation to
+     * @param annotation  the annotation to add
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToConstructor(Constructor<T> constructor, Annotation annotation);
+
+    /**
+     * Add an annotation to the specified constructor. If the constructor is not
+     * already present, it will be added.
+     *
+     * @param constructor the constructor to add the annotation to
+     * @param annotation  the annotation to add
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToConstructor(AnnotatedConstructor<T> constructor, Annotation annotation);
+
+    /**
+     * Remove an annotation from the specified constructor.
+     *
+     * @param constructor    the constructor to add the annotation to
+     * @param annotationType the annotation to add
+     * @throws IllegalArgumentException if the annotationType is null or if the
+     *                                  constructor is not currently declared on the type
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructor(Constructor<T> constructor,
+                                                       Class<? extends Annotation> annotationType);
+
+    /**
+     * Remove an annotation from the specified constructor.
+     *
+     * @param constructor    the constructor to add the annotation to
+     * @param annotationType the annotation to add
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  annotation does not exist on the type or if the constructor is
+     *                                  not currently declared on the type
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructor(AnnotatedConstructor<T> constructor,
```

Comment:
I'm not saying it's wrong (my generic sience is limited ;) ). But most of these methods come from DeltaSpike (https://github.com/apache/deltaspike/blob/master/deltaspike/core/api/src/main/java/org/apache/deltaspike/core/util/metadata/builder/AnnotatedTypeBuilder.java#L341)

So can you double check, because that would mean DS as a bug too.